content/blogposts/2024/codyception-coding-cody-tools-with-cody.md

```diff
@@ -0,0 +1,159 @@
+
+![](http://cody.kynlo.co.uk/codykyn.png)
+
+Now, nobody is forcing you *hides the pool noodle* to use my little tool, and you are absolutely free to go create these manually!  There's [highly detailed documentation](https://sourcegraph.com/notebooks/Tm90ZWJvb2s6MzA1NQ==#custom-commands-194e6b3f-f682-475f-9c66-cfcc84d05c66) written by the amazing [beatrix](https://twitter.com/3eatrix), for those who want to dive even deeper into all of that beautiful behind-the-scenes code!
+
+
+## Examples of Time-Saving Commands
+
+![](http://cody.kynlo.co.uk/2.png)
+
+```json
+    "implement-code-localization-support": {
+      "description": "implement-code-localization-support",
+      "prompt": "Implement code localization support in the current project, enabling adaptation to different languages and cultural preferences.",
+      "context": {
+        "codebase": true
+      }
+    },
+    "check-code-code-smells": {
+      "description": "check-code-code-smells",
+      "prompt": "Check and identify code smells in the current project, suggesting refactoring for improved code quality.",
+      "context": {
+        "codebase": true
+      }
+    },
+    "evaluate-code-external-apis": {
+      "description": "evaluate-code-external-apis",
+      "prompt": "Evaluate and report on the usage and reliability of external APIs in the current project, suggesting updates or replacements.",
+      "context": {
+        "codebase": true
+      }
+    },
+    "implement-code-database-indexing": {
+      "description": "implement-code-database-indexing",
+      "prompt": "Implement code database indexing strategies in the current project, optimizing data retrieval and query performance.",
+      "context": {
+        "codebase": true
+      }
+```
+
+## The cody.json File: Tailoring Commands to Your Needs ##
+
+Customizing commands is made easy with the `cody.json` file. Here's how you can leverage it:
+
+### Project-Specific Commands:
+
+Create `cody.json` in the `.vscode` directory of your project to make commands accessible only in that workspace. Cody will execute these custom commands when working on the project.
+
+### Global commands:
+
+Create a new `cody.json` file in the `.vscode` directory of your home directory for commands accessible across multiple projects. Cody will be able to execute these global custom commands in any workspace.
+
+### Workspace vs. User Commands
+
+Understanding the scope of your custom commands is essential:
+
+- `cody.json` in a project's `.vscode` folder: Commands are available only in that workspace.
+- `cody.json` in the home `.vscode` folder: Commands are available globally across all workspaces.
+
+This flexibility allows you to organize project-specific commands separately from reusable commands accessible globally, enhancing your customization capabilities with Cody.
+
+## Cody's custom commands syntax
```

Comment:
Was there supposed to be a paragraph or more content here?

Reply:
That was a left over, now removed